src/app/data-planes/data/index.ts

```diff
@@ -208,20 +200,68 @@ export const SidecarDataplane = {
     }
   },
 }
-
-export const InspectRules = {
-  fromCollection(partialInspectRules: PartialInspectRulesForDataplane): InspectRulesForDataplane & {} {
-    const rules = Array.isArray(partialInspectRules.rules) ? partialInspectRules.rules : []
-    const proxyRules = getProxyRules(rules)
-    const toRules = getToRules(rules)
-    const fromRuleInbounds = getFromRuleInbounds(rules)
-
+export type Rule = Omit<InspectBaseRule, 'conf' | 'origin'> & {
+  type: string
+  ruleType: string
+  inbound?: InspectInbound
+  config: InspectBaseRule['conf']
+  origins: InspectBaseRule['origin']
+}
+export type RuleCollection = Omit<PartialInspectRulesForDataplane, 'rules'> & {
+  // httpMatches: unknown[]
+  rules: Rule[]
+}
+export const Rule = {
```

Comment:
**Comment**: The naming in this area of `/_rules` APIs, target refs, and policy matching unfortunately has two places where the name "rule" appears. We have what you call `Rule` here which are the top-level elements of the `/_rules` API’s `rules` array. However, further down in the response of the `/_rules` API are _different_ `rules` arrays (it’s these rules that we spoke about in yesterday’s design sync meeting). I _think_ it would be a good idea to use a more specific name here to disambiguate. In the type definitions for this, I’ve used `InspectRule` because the name "Inspect API" is sometimes used to describe this API, but I’m not sure if that’s any good as a name.

Reply:
If I follow what you are saying here, this has all been flattened out now in this PR. The reshaping here removes the top-level/further down structure and you just end up with one big list of `Rules`.

Let me know if I haven't followed what you were saying, I was wondering whether I could simply call these `Rules` and it looked at the time that I could.

Reply:
Let’s move on with that. I’m still a little concerned with this "taking up" the term rule here as these rules can appear elsewhere, but we can worry about that then.